app/src/main/java/vn/sun/music_68/data/source/remote/LoadTrackAsync.java

```diff
@@ -0,0 +1,48 @@
+package vn.sun.music_68.data.source.remote;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.sun.music_68.base.BaseAsyncTask;
+import vn.sun.music_68.data.model.Track;
+import vn.sun.music_68.data.source.TrackDataSource;
+
+public class LoadTrackAsync extends BaseAsyncTask<Track> {
+    private static final String TRACK = "track";
+    private static final String TITLE = "title";
+    private static final String KEY_USER = "user";
+    private static final String KEY_USER_USERNAME = "username";
+    private static final String ID = "id";
+    private static final String ARTWORK_URL = "artwork_url";
+    private static final String COLLECTION = "collection";
+
+    public LoadTrackAsync(TrackDataSource.DataCallback<Track> callback) {
+        super(callback);
+    }
+
+    @Override
+    public List<Track> convertJSON(String jsonString) {
+        List<Track> trackList = new ArrayList<>();
+        try {
+            JSONObject result = new JSONObject(jsonString);
+            JSONArray collection = new JSONArray(COLLECTION);
+            for (int index = 0; index < collection.length(); index++) {
+                JSONObject trackInfo = collection.getJSONObject(index);
+                JSONObject track = trackInfo.getJSONObject(TRACK);
+                String title = track.getString(TITLE);
+                String artworkUrl = track.getString(ARTWORK_URL);
+                int id = track.getInt(ID);
+                String trackArtist = track.getJSONObject(KEY_USER).getString(KEY_USER_USERNAME);
+                Track object = new Track(id, title, trackArtist);
```

Comment:
Đoạn này em nghĩ viết một hàm constructor trong Track với tham số truyền vào là JSONObject thì sẽ gọn hơn nhiều 
Hơn nữa mấy cái trường TITLE, ARTWORK,... thuộc về Track chứ k nên thuộc về AsyncTask 
```suggestion
                JSONObject trackJson = ....
                Track track = new Track(trackJson);
```